internal/broker/broker.go

```diff
@@ -735,3 +736,13 @@ func decorateErrorMessage(data *isAuthenticatedDataResponse, msg string) {
 	errMsg.Message = fmt.Sprintf("%s: %s", msg, errMsg.Message)
 	*data = errMsg
 }
+
+// Checks if the provided error is of type ForDisplayError. If it is, it returns the error message. Else, it returns
+// the provided fallback message.
+func errorMessageForDisplay(err error, fallback string) errorMessage {
```

Comment:
Just a note while I am on this: I think we should take this (as explained on the uncrypted token PR) as a way to logging in the initial error and returning the error we want to display to the user (probably on the defer of each dbus calls). I wonder if all those redacted messages are always coming from the provider? If so, it makes sense to have them as a subpackage of providers), otherwise, we can move it directly under `internal/`. To not conflict with the "errors" I would just name it to something similar you imported it with: "providerserrors" or even directly "errorfordisplay" as this is exactly what's its scope is.

We can go on this intermediate step and fix this in the token PR too if you prefer.

Reply:
As told on the error one, I’m happy to give a pass on that one, and I’ll refactor it later this pulse, hoping I’m getting time for it.